doc/query.md

```diff
@@ -94,11 +150,60 @@ if (!name) {
 }
 ```
 
-`-ensureIndexed:withName:` returns the name of the index if it is successful,
-otherwise it returns `nil`.
+####Indexing for text search
+
+Since text search relies on SQLite FTS which is a compile time option, we must ensure that SQLite FTS is in fact available.  To verify that text search is enabled and that a text index can be created use `-isTextSearchEnabled` before attempting to create a text index.  If text search is not enabled see [compiling and enabling SQLite FTS][enableFTS] for details. 
+
+[enableFTS]: http://www.sqlite.org/fts3.html#section_2
+
+```objc
+if ([ds isTextSearchEnabled]) {
+    // Create a text index over the name and comment fields.
+    NSString *name = [ds ensureIndexed:@[@"name", @"comment"]
+                              withName:@"basic_text_index"
+                                  type:@"text"];
+    if (name == nil) {
+        // there was an error creating the index
+    }
+}
+```
+
+As text indexing relies on SQLite FTS functionality any custom tokenizers need to be managed through SQLite.  SQLite comes standard with the "simple" default tokenizer as well as a Porter stemming algorithm tokenizer ("porter").  Please refer to [SQLite FTS tokenizers][fts] for additional information on custom tokenizers.
+
+[fts]: http://www.sqlite.org/fts3.html#tokenizer  
+
+When creating a text index, overriding the default tokenizer setting is done by providing a `tokenize` parameter setting as part of the index settings.  The value should be the same as the tokenizer name given to SQLite when registering that tokenizer.  In the example below we set the tokenizer to `porter`.
```

Comment:
I wonder if the sentence "The value should be the same as the tokenizer name given to SQLite when registering that tokenizer." should just be dropped. It was a little confusing for me until I realized that "registering" a tokenizer was outside of the CDTDatastore stuff.  Also, is it possible for a developer to build custom tokenizers and install them in SQLite on a customer's phone/ipad? (Same question for the android stuff).